File: src/main/java/pl/training/workshop/m6/s17_singleton/step1/TicketDesk.java
package pl.training.workshop.m6.s17_singleton.step1;

import pl.training.workshop.shared.Money;

/** Krok 1: klient pyta o jedyną instancję zamiast robić new. */
public final class TicketDesk {
    public Money quote(String format, boolean online) {
        Money price = PriceList.getInstance().basePrice(format);
        return online ? price.plus(Money.of("2.00")) : price;
    }
}

File: typescript/src/workshop/m6/s17_singleton/step1/TicketDesk.ts
import { Money } from '../../../shared/Money.js';
import { PriceList } from './PriceList.js';

/** Krok 1: klient pyta o jedyną instancję zamiast robić new. */
export class TicketDesk {
  quote(format: string, online: boolean): Money {
    const price = PriceList.getInstance().basePrice(format);
    return online ? price.plus(Money.of('2.00')) : price;
  }
}
